### DISCUSSION TITLE
How can I sign an etherium message?

### DISCUSSION THREAD

**Discussion by VovaLobanov** [trustwallet/wallet-core#2545] [Q&A]
In the documentation and tests I can see how can I sign a transaction. (using EthereumSigningInput and EthereumSigningOutput, AnySigner.sign(input: signerInput, coin: .ethereum)).
But how to handle message signing?
Thanks in advance

**Comment by optout21** [comment]
It is possible to sign a message/hash using `PrivateKey` directly:
https://github.com/trustwallet/wallet-core/blob/master/include/TrustWalletCore/TWPrivateKey.h#L119
